plugins/processors/ec2tagger/ec2tagger_test.go

```diff
@@ -206,43 +214,35 @@ func (m *mockEC2Metadata) GetInstanceIdentityDocument() (ec2metadata.EC2Instance
 
 func TestInitFailWithNoMetadata(t *testing.T) {
 	assert := assert.New(t)
-	mockMetadata := &mockEC2Metadata{
-		IsAvailable:              false,
-		InstanceIdentityDocument: nil,
-	}
 
-	tagger := Tagger{
-		Log:         testutil.Logger{},
-		ec2metadata: mockMetadata,
-	}
-	err := tagger.Init()
-	assert.NotNil(err)
-	assert.Contains(err.Error(), "ec2tagger: Unable to retrieve InstanceId. This plugin must only be used on an EC2 instance")
-}
-
-func TestInitFailWithNoInstanceIdentityDocument(t *testing.T) {
```

Comment:
For the background a little bit: With the last EC2 Tagger, it uses [Available and GetInstanceIdentityDocument](https://github.com/aws/amazon-cloudwatch-agent/pull/480/files#diff-6638b992018408bf44cac12361171fd050ff44fcc24c28f326988f4768a6dc2dL319-L329). However, under the hood, `Available` only check if its able to get Instance-ID](https://github.com/aws/aws-sdk-go/blob/main/aws/ec2metadata/api.go#L216), it only improves slightly performance when comparing to `GetInstanceIdentityDocument`. However, it also provides drawback when CloudWatchAgent retries to scrap IMDS (extra 4 second with 1 retry/ second) and also without retry ( 1 extra second). Therefore, in the new PR, I only cover [GetInstanceIdentityDocument](https://github.com/aws/amazon-cloudwatch-agent/pull/480/files#diff-6638b992018408bf44cac12361171fd050ff44fcc24c28f326988f4768a6dc2dR256-R261). 

To main point, we have a test which covers [Instance Identity Document as nil](https://github.com/aws/amazon-cloudwatch-agent/blob/9adb253817dedb4d27a5a5e1f7329beb3465eb25/plugins/processors/ec2tagger/ec2tagger_test.go#L215-L233). Therefore, I would consider the mentioned test rather useless since its a duplication with the aforementioned test.